pkg/apis/monitoring/v1/types.go

```diff
@@ -381,7 +388,10 @@ type PrometheusSpec struct {
 	// Time duration Prometheus shall retain data for. Default is '24h' if
 	// retentionSize is not set, and must match the regular expression `[0-9]+(ms|s|m|h|d|w|y)`
 	// (milliseconds seconds minutes hours days weeks years).
-	Retention string `json:"retention,omitempty"`
+	// Time based retention can be disabled by setting Retention to `0` but it requires
+	// a non-empty size-based retention to be effective
+	// +kubebuilder:default:="24h"
```

Comment:
I realize that we can't handle the default value at this level: if `retentionSize` is not empty, the operator shouldn't set `retention` to 24h...